ipn/ipnlocal/local.go

```diff
@@ -4463,6 +4463,9 @@ func (b *LocalBackend) setTCPPortsInterceptedFromNetmapAndPrefsLocked(prefs ipn.
 	if prefs.Valid() && prefs.RunSSH() && envknob.CanSSHD() {
 		handlePorts = append(handlePorts, 22)
 	}
+	if b.ShouldRunWebClient() {
```

Comment:
This method is wrong. It access `b.netMap` without a lock held:

```go
func (b *LocalBackend) ShouldRunWebClient() bool {
	return b.webclientAtomicBool.Load() && hasCapability(b.netMap, tailcfg.CapabilityPreviewWebClient)
}
```

Reply:
Maybe it should set an atomic whenever `b.netMap` changes and not try to linear scan `b.netMap`'s capabilities (and potentially acquire a lock) whenever that question needs to be asked. (not sure how often)

Reply:
Ack, @willnorris maybe we should have `webclientAtomicBool` be the source of truth, and updated when the cap changes?

Reply:
Yeah that's what I'm thinking

Reply:
Fixed in b370274b29529b308c55de108b0c6156586fc632

Reply:
ShouldRunWebClient should probably be documented that it's safe to call with either b.mu held or not. 

Because otherwise this still looks shady, calling an exported method from an unexported "Locked"-suffixed method.

Reply:
Added docs